operations/rollout-operator/rollout-operator.libsonnet

```diff
@@ -0,0 +1,380 @@
+(import 'ksonnet-util/kausal.libsonnet') +
+
+(import 'images.libsonnet') +
+
+(import 'rollout-operator-utils.libsonnet') +
+
+// Support for ReplicaTemplate objects.
+(import 'replica-template.libsonnet') +
+
+// Support for ZoneAwarePodDisruptionBudget objects.
+(import 'zone-aware-pod-disruption-budget.libsonnet') +
+
+{
+  local clusterRole = $.rbac.v1.clusterRole,
+  local clusterRoleBinding = $.rbac.v1.clusterRoleBinding,
+  local container = $.core.v1.container,
+  local deployment = $.apps.v1.deployment,
+  local mutatingWebhook = $.admissionregistration.v1.mutatingWebhook,
+  local mutatingWebhookConfiguration = $.admissionregistration.v1.mutatingWebhookConfiguration,
+  local policyRule = $.rbac.v1.policyRule,
+  local role = $.rbac.v1.role,
+  local roleBinding = $.rbac.v1.roleBinding,
+  local service = $.core.v1.service,
+  local serviceAccount = $.core.v1.serviceAccount,
+  local servicePort = $.core.v1.servicePort,
+  local validatingWebhook = $.admissionregistration.v1.validatingWebhook,
+  local validatingWebhookConfiguration = $.admissionregistration.v1.validatingWebhookConfiguration,
+
+  _config+:: {
+    rollout_operator_enabled: false,
+
+    // Configure the rollout operator to accept webhook requests made as part of scaling
+    // statefulsets up or down. This allows the rollout operator to ensure that stateful
+    // components (ingesters, store-gateways) are scaled up or down safely.
+    enable_rollout_operator_webhook: false,
+
+    // Specifically enable zone aware pod distruption budget support, including the rollout-operator webhooks.
+    // Note that zpdb support is enabled by default if webhooks are enabled.
+    // See disable_rollout_operator_eviction_webhook for additional configuration options.
+    enable_zpdb: false,
```

Comment:
Should these default to true?

Reply:
I would have them default to enabled here, and then have Mimir override the defaults to suit Mimir.